Tests/SkipFoundationTests/Foundation/LocaleTests.swift

```diff
@@ -727,6 +727,197 @@ final class LocaleTests: XCTestCase {
     static let checkLocaleCodes: [String] = [
         "ar", "ca", "zh-HK", "zh-Hant", "zh-Hans", "hr", "cs", "da", "nl", "en-US", "en-AU", "en-IN", "en-GB", "fi", "fr-CA", "de", "el", "he", "id", "it", "ja", "ko", "ms", "nb", "pl", "pt-BR", "pt-PT", "ro", "ru", "sk", "es", "es-419", "sv", "th", "tr", "uk", "vi"
     ]
+
+    func testCommonISOCurrencyCodes() throws {
+        let commonISOCurrencyCodes = Locale.commonISOCurrencyCodes
+        let checkCurrencyCodes = Self.checkCurrencyCodes.keys.sorted()
+
+        XCTAssertEqual(commonISOCurrencyCodes, checkCurrencyCodes)
+    }
+
+    func testCurrencyCodes() throws {
+        for localeCode in Self.checkLocaleCodes {
+            let locale = Locale(identifier: localeCode)
+
+            XCTAssertNil(locale.localizedString(forCurrencyCode: "INVALID"))
+
+            if localeCode == "en-US" {
+                for (currencyCode, localizedString) in Self.checkCurrencyCodes {
+                    XCTAssertEqual(localizedString, locale.localizedString(forCurrencyCode: currencyCode))
+                    XCTAssertEqual(localizedString, locale.localizedString(forCurrencyCode: currencyCode.lowercased()))
+                }
+            } else {
+                for currencyCode in Self.checkCurrencyCodes.keys {
+                    let localizedStringCamelCase = locale.localizedString(forCurrencyCode: currencyCode)
```

Comment:
The variable name is a typo. I decided against supporting case insensitivity for the Android Locale instance as it's not exactly API compatible with Swift.